### PR TITLE
[tlse] tls for NeutronAPI pod configuration

### PR DESCRIPTION
**Pull request by stuggi**
Public/Internal service cert secrets and the CA bundle secret can be passed to configure httpd virtual hosts for tls termination. The certs get direct mounted to the appropriate place in etc/pki/tls/certs/%s.crt|key and a CA bundle to
/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem . Job deployments for bootstrap/cron get the CA bundle added if configured.

Also indexes the named input resources for password, CA bundle, and endpoint secrets to be able to watch them for any changes and reconcile if needed.

Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/428

Jira: [OSPRH-2197](https://issues.redhat.com//browse/OSPRH-2197)

### CODE REVIEW

**Comment by stuggi** [comment]
recheck

File: templates/neutronapi/httpd/10-neutron-httpd.conf [hunk 1]
```diff
@@ -1,4 +1,8 @@
+{{ range $endpt, $vhost := .VHosts }}
+# {{ $endpt }} vhost {{ $vhost.ServerName }} configuration
 <VirtualHost *:9696>
```
Comment (slawqo) [inline templates/neutronapi/httpd/10-neutron-httpd.conf:3]: shouldn't we use port 13696 if TLS is enabled? IIRC this is what we are using in e.g. OSP 17.1

Comment (stuggi) [reply templates/neutronapi/httpd/10-neutron-httpd.conf:3]: why you think we should do this? its just an internal port on the httpd behind a k8s service.

Comment (slawqo) [reply templates/neutronapi/httpd/10-neutron-httpd.conf:3]: ok, so in such case we don't need to do it like that. You are right

**Review by slawqo — APPROVED** [review]

**Comment by stuggi** [comment]
@slawqo I have added some more tests and rebased it. can you please restore the approval?

**Review by karelyatin — APPROVED** [review]

**Pull request merged**